src/app/services/auth/auth.service.ts

```diff
@@ -1,106 +1,38 @@
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
 
 @Injectable()
 export class AuthService {
-  apiBaseUrl;
-  COOKIE_KEYS;
-  user;
   constructor(
-    private cookieService: CookieService,
-    private appConfigService: AppConfigService,
-    private environmentService: EnvironmentService
+    private readonly httpService: HttpClient
   ) {
-    this.COOKIE_KEYS = {
-      TOKEN: config.cookies.token,
-      USER: config.cookies.userId,
-      ROLES: config.cookies.roles,
-    };
-
-    this.user = null;
-  }
-
-  canActivate() {
-    console.log('reached can activate');
-    if (!this.isAuthenticated()) {
-      this.loginRedirect();
-      return false;
-    }
-
-    return true;
-  }
-
-  /**
-   * Generate the Idam Login url.
-   */
-  generateLoginUrl() {
-
-    const SCOPE = `profile openid roles manage-user create-user`;
-
-    return this.environmentService.config$.map( environmentConfig => {
-
-      const { clientId, idamWeb, oAuthCallback, protocol } = environmentConfig;
-
-      const port = window.location.port ? `:${window.location.port}` : ``;
-      const API_BASE_URL = `${protocol}://${window.location.hostname}${port}`;
-
-      const callback = `${API_BASE_URL}/${oAuthCallback}`;
-
-      return `${idamWeb}/login?response_type=code&client_id=${clientId}&redirect_uri=${callback}&scope=${SCOPE}`;
-    });
-  }
-
-  loginRedirect() {
-    this.generateLoginUrl().subscribe( url => {
-      window.location.href = url;
-    });
   }
 
-  decodeJwt(jwt) {
-    return jwtDecode(jwt);
+  public loginRedirect() {
+    const href = '/auth/login';
+    this.setWindowLocationHref(href);
   }
 
-
-  isAuthenticated(): boolean {
-    const jwt = this.cookieService.get(this.COOKIE_KEYS.TOKEN);
-    if (!jwt) {
-      return false;
-    }
-    const jwtData = this.decodeJwt(jwt);
-    const notExpired = jwtData.exp > Math.round(new Date().getTime() / 1000);
-    // do stuff!!
-    return notExpired;
+  public isAuthenticated(): Observable<boolean> {
+    return this.httpService.get<boolean>('/auth/isAuthenticated');
   }
 
-  signOut() {
-
-    window.location.href = '/api/logout';
+  public signOut() {
+    const href = '/auth/logout';
+    this.setWindowLocationHref(href);
   }
 
-  logOutAndRedirect() {
-
-    console.log('logOutAndRedirect');
-    const { hostname, port } = this.getWindowLocation();
-    const portNumber = port ? `:${port}` : '';
-
-    const baseUrl = `http://${hostname}${portNumber}`;
-
-    const idleSignOutUrl = `${baseUrl}/idle-sign-out`;
-    const href = `api/logout?redirect=${idleSignOutUrl}`;
-    this.setWindowLocationHref(href);
+  public logOut(): Observable<any> {
+    return this.httpService.get('/auth/logout?noredirect=true');
   }
 
-  public getWindowLocation() {
-    return window.location;
+  public logOutAndRedirect() {
+    this.logOut().subscribe( () => {
+      this.setWindowLocationHref('/idle-sign-out')
+    })

```

Comment:
missing ; which has tslint issue

Reply:
fixed, though it would only have given a warning (which we have plenty of anyway :joy: )